TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/EditInduction/CompletedDate.cshtml.cs

```diff
@@ -1,30 +1,74 @@
+using System.ComponentModel.DataAnnotations;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using TeachingRecordSystem.Core.DataStore.Postgres;
 
 namespace TeachingRecordSystem.SupportUi.Pages.Persons.PersonDetail.EditInduction;
 
 [Journey(JourneyNames.EditInduction), ActivatesJourney, RequireJourneyInstance]
 public class CompletedDateModel : CommonJourneyPage
 {
+    protected TrsDbContext _dbContext;
+    protected IClock _clock;
+
+    protected InductionStatus InductionStatus => JourneyInstance!.State.InductionStatus;
+    public string? PersonName { get; set; }
+
+    [BindProperty]
+    [DateInput(ErrorMessagePrefix = "Completed date")]
+    [Required(ErrorMessage = "Enter an induction completed date")]
+    [Display(Name = "When did they complete induction?")]
+    public DateOnly? CompletedDate { get; set; }
+
     public InductionJourneyPage NextPage => InductionJourneyPage.ChangeReasons;
+
     public string BackLink
     {
-        // TODO - more logic needed when other routes to completed-date are added
-        get => PageLink(InductionJourneyPage.StartDate);
+        get
+        {
+            return JourneyInstance!.State.JourneyStartPage == InductionJourneyPage.CompletedDate
+                ? LinkGenerator.PersonInduction(PersonId)
+                : PageLink(InductionJourneyPage.StartDate);
+        }
     }
 
-    public CompletedDateModel(TrsLinkGenerator linkGenerator) : base(linkGenerator)
+    public CompletedDateModel(TrsLinkGenerator linkGenerator, TrsDbContext dbContext, IClock clock) : base(linkGenerator)
     {
+        _dbContext = dbContext;
+        _clock = clock;
     }
 
-    public void OnGet()
+    public Task OnGetAsync()
     {
+        CompletedDate = JourneyInstance!.State.CompletedDate;
+        return JourneyInstance!.UpdateStateAsync(state =>
```

Comment:
Shouldn't be doing updates in a Get method .. and initial state setting should be done in `EnsureInitializedAsync`